src/hypothesis/extra/pandas/impl.py

```diff
@@ -0,0 +1,617 @@
+        self.dtype = dtype
+        self.min_size = min_size
+        self.max_size = max_size
+        self.unique = unique
+
+    def do_draw(self, data):
+        result = []
+        seen = set()
+
+        iterator = cu.many(
+            data, min_size=self.min_size, max_size=self.max_size,
+            average_size=(self.min_size + self.max_size) / 2
+        )
+
+        while iterator.more():
+            elt = data.draw(self.elements)
+
+            if self.unique:
+                if elt in seen:
+                    iterator.reject()
+                    continue
+                seen.add(elt)
+            result.append(elt)
+
+        dtype = infer_dtype_if_necessary(
+            dtype=self.dtype, values=result, elements=self.elements,
+            draw=data.draw
+        )
+        return pandas.Index(result, dtype=dtype)
+
+
+DEFAULT_MAX_SIZE = 10
+
+
+@st.cacheable
+@st.defines_strategy
+def range_indexes(min_size=0, max_size=None):
+    st.check_valid_interval(min_size, max_size, 'min_size', 'max_size')
+    if max_size is None:
+        max_size = min_size + DEFAULT_MAX_SIZE
+    return st.integers(min_size, max_size).map(
+        lambda i: pandas.Index(hrange(i), dtype=int)
+    )
+
+
+@st.cacheable
+@st.defines_strategy
+def indexes(
+    elements=None, dtype=None, min_size=0, max_size=None, unique=True,
+):
+    """Provides a strategy for producing a :class:`pandas.Index`.
+
+    * elements is a strategy which will be used to generate the individual
+      values of the index. If None, it will be inferred from the dtype.
+    * dtype is the dtype of the resulting index. If None, it will be inferred
+      from the elements strategy. At least one of dtype or elements must be
+      provided.
+    * min_size is the minimum number of elements in the index.
+    * max_size is the maximum number of elements in the index. If None then it
+      will default to a suitable small size. If you want larger indexes you
+      should pass a max_size explicitly.
+    * unique specifies whether all of the elements in the resulting index
+      should be distinct.
+
+    """
+    st.check_valid_interval(min_size, max_size, 'min_size', 'max_size')
+    st.check_type(bool, unique, 'unique')
+
+    elements, dtype = elements_and_dtype(elements, dtype)
+
+    if max_size is None:
+        max_size = min_size + DEFAULT_MAX_SIZE
+    return ValueIndexStrategy(
+        elements, dtype, min_size, max_size, unique)
+
+
+@st.defines_strategy
+def series(elements=None, dtype=None, index=None, fill=None, unique=False):
+    """Provides a strategy for producing a :class:`pandas.Series`.
+
+    Arguments:
+
+        * elements: a strategy that will be used to generate the individual
+          values in the series. If None, we will attempt to infer a suitable
+          default from the dtype.
+
+        * dtype: the numpy.dtype of the resulting series and may be any value
+          that can be passed to numpy.dtype. If None, will use `pandas'
```

Comment:
that can be passed to :class:`numpy.dtype`. If None, uses Pandas'

(cross-reference and fixes formatting of 'Pandas')